kong/plugins/acme/client.lua

```diff
@@ -13,6 +13,84 @@ local CERTKEY_KEY_PREFIX = "kong_acme:cert_key:"
 
 local LOCK_TIMEOUT = 30 -- in seconds
 
+local function get_allowed_config_domains(host)
+  local domains = {}
+  local t = ngx.re.match(host, "(?:(.+)[.])*([^.]+[.][^.]+)$", "jo")
+
+  -- Root domain wildcard pattern
+  local root = t[table.maxn(t)]
+  table.insert(domains, root)
+  table.insert(domains, "*." .. root)
+
+  if t[1] then
+    local it, err = ngx.re.gmatch(t[1], "([^.]+)", "i")
+    local subs = {}
+    while true do
+      local m, err = it()
+      if not m then
+        break
+      end
+
+      if not err then
+        table.insert(subs, m[0])
+      end
+    end
+
+    local prev = root
+    for i=1, #subs do
+      local index = #subs + 1 - i
+      local new_domain = subs[index] .. "." .. prev
+
+      if index == 1 then
+        table.insert(domains, new_domain)
+
+      else
+        table.insert(domains, "*." .. new_domain)
+      end
+
+      prev = new_domain
+    end
+  end
+
+  return domains
+end
+
+
+local function load_domain(domain)
+  kong.log.debug("Loading domain from DB ", domain)
+  local entity, err = kong.db.acme_domain:select_by_name(domain)
+  if not entity then
+    return nil, err
+  end
+  return entity
+end
+
+
+local function is_domain_db_config_exists(host)
+  local allowed_domains = get_allowed_config_domains(host)
+  for _, domain in ipairs(allowed_domains) do
+    kong.log.debug("checking domain " .. domain .. " in DB")
+    local domain_cache_key = kong.db.acme_domain:cache_key(domain)
+    local entity, err = kong.cache:get(domain_cache_key, nil, load_domain, domain)
+
+    if err then
+      kong.log.err("can't load domain from storage: ", err)
+      --return kong.response.exit(500, { message = "Unexpected error" })
+      -- Not sure if domain exists, return true?
```

Comment:
this initially was `return kong.response.exit(500, { message = "Unexpected error" })` 

But now since this function is being used in renewal we cannot simply return any http response and stop the loop.

Therefore, if there is an error while looking for host in db/cache, we should assume the best (i.e it is present in db) and move on with certificate generation / renewal.